tools/docker/Dockerfile

```diff
@@ -142,7 +142,8 @@ FROM debian:stretch as firecracker-vm-root-builder
 COPY --from=firecracker-vm-root / /vm
 RUN mkdir -p /output \
 	&& cd /output \
-	&& mkfs.ext4 -d /vm vm.ext4 65536
+	&& mkfs.ext4 -d /vm vm.ext4 65536 \
+	&& chown 0666 vm.ext4
```

Comment:
Any files provided to the jail need to be readable, another options was to change the `uid` and `gid` of the file, but that didn't seem like the behaviour we'd want.

Reply:
Ok, but mode 0666 does much more than make the file readable.

1. It makes it writable
2. It makes it read/writable for *everybody that can see it*

If we're going to copy files around, then we should be setting owner and group bits to restrict them to the specific uid/gid needed, and we should be using the most restrictive permissions we can get away with. If we're going to bind mount files, then we should make sure that the mount is read-only.

Rather than setting file modes, you might consider using [ACLs](https://manpages.debian.org/buster/acl/setfacl.1.en.html)

Reply:
Turns out we didn't need to do this. I was running into permission issues earlier because I didn't call the `chownr` function I had created. But it looks like this is no longer needed.